src/schemes/structure/total_lagrangian_sph/system.jl

```diff
@@ -137,35 +137,40 @@ function TotalLagrangianSPHSystem(initial_condition, smoothing_kernel, smoothing
         @assert allunique(clamped_particles) "`clamped_particles` contains duplicate particle indices"
 
         n_clamped_particles = length(clamped_particles)
-        move_particles_to_end!(initial_condition, collect(clamped_particles))
-        move_particles_to_end!(young_modulus, collect(clamped_particles))
-        move_particles_to_end!(poisson_ratio, collect(clamped_particles))
+        ic_sorted_indices = deepcopy(initial_condition)
```

Comment:
```suggestion
        initial_condition_sorted = deepcopy(initial_condition)
```
I think that's easier to understand in the code below.